File: api/src/main/java/org/egglog/api/worktype/model/dto/response/WorkTypeResponse.java
package org.egglog.api.worktype.model.dto.response;

import lombok.*;
import org.egglog.api.worktype.model.entity.WorkTag;

import java.time.LocalTime;

/**
 * ```
 * ===================[Info]=========================
 * packageName    : org.egglog.api.worktype.model.dto.response
 * fileName      : WorkTypeResponse
 * description    : 근무 타입 응답 객체
 * =================================================
 * ```
 * |DATE|AUTHOR|NOTE|
 * |:---:|:---:|:---:|
 * |2024-04-29|김형민|최초 생성|
 */
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class WorkTypeResponse {
    private Long workTypeId;
    private String title;
    private String color;       //색상
    private String workTypeImgUrl;      //근무 이미지
    private WorkTag workTag; //Day, Eve, Night, Off, ETC 태그 속성
    private LocalTime startTime;       //시작시간
    private LocalTime workTime;     //종료 시간
}

File: client/data/src/main/java/com/org/egglog/data/myCalendar/model/WorkTypeResponse.kt
package com.org.egglog.data.myCalendar.model

import com.org.egglog.domain.myCalendar.model.WorkType
import kotlinx.serialization.Serializable

@Serializable
data class WorkTypeResponse (
    val workTypeId: Int,
    val title: String,
    val color: String,
    val workTypeImgUrl: String,
    val workTag: String,
    val startTime: String,
    val workTime: String
)

fun WorkTypeResponse.toDomainModel(): WorkType {
    return WorkType(workTypeId, title, color, workTypeImgUrl, workTag, startTime, workTime)
}